app/src/main/java/com/selfdrivingboat/boatcontroller/MainActivity.java

```diff
@@ -1,394 +1,445 @@
+import android.view.View;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
+import com.android.volley.RequestQueue;
+import com.datadog.android.log.Logger;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+
+import org.json.JSONException;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static android.hardware.SensorManager.DATA_X;
+import static android.hardware.SensorManager.DATA_Y;
+import static android.hardware.SensorManager.DATA_Z;
+
+
+public class MainActivity extends AppCompatActivity implements OnBluetoothDeviceClickedListener,SensorEventListener {
+    private final int REQUEST_PERMISSION_ACCESS_FINE_LOCATION = 1;
+
+
+    private static final int REQUEST_CONNECT = 1;
+    private static final int PERMISSION_REQUEST_COARSE_LOCATION = 2;
+    public static final String EXTRAS_DEVICE_NAME = "extras_device_name";
+    public static final String EXTRAS_DEVICE_ADDRESS = "extras_device_address";
+    public String mConnectionState = BluetoothLeService.ACTION_GATT_DISCONNECTED;
+
+    private static final int REQUEST_ENABLE_BT = 1;
+    private static final long SCAN_PERIOD = 1000 * 10;
+    private SwipeRefreshLayout swipeRefresh;
+    private RecyclerView recyclerView;
+    private MyBluetoothDeviceAdapter mBluetoothDeviceAdapter;
+    private List<BluetoothDevice> mBluetoothDeviceList = new ArrayList<>();
+    private MyBluetoothScanCallBack mBluetoothScanCallBack = new MyBluetoothScanCallBack();
+    private Handler mHandler;
+    private BluetoothLeService mBluetoothLeService;
+    private String mDeviceName;
+    private String mDeviceAddress;
+    public Logger logger = DatadogLogger.getInstance();
+    private SelfDriving selfDriving = new SelfDriving();
+    public FusedLocationProviderClient fusedLocationClient;
+
+    public RequestQueue volleyQueue;
+    private SensorManager sensorManager;
+    private Sensor mACCELEROMETER;
+    public float  acceleRometer_x,  acceleRometer_y,  acceleRometer_z;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+        mACCELEROMETER = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        sensorManager.registerListener(this,mACCELEROMETER, SensorManager.SENSOR_DELAY_NORMAL);

```

Comment:
this lines breaks the app.
If you comment it out you will see that everything works fine, if you leave it the app break. 
can't find any stack trace just find "BoatController isn'responding"

wdyt @mack-4567 ?